### ISSUE TITLE
Unknown value '571' was given for enum SwedbankPaymentPortal\SharedEntity\Type\PurchaseStatus

### ISSUE THREAD

**Issue by jsakars** [Swedbank-SPP/swedbank-payment-portal#10]
`Unknown value '571' was given for enum SwedbankPaymentPortal\SharedEntity\Type\PurchaseStatus` RuntimeException is being thrown when calling `handlePendingTransaction` on `PayPalService` class.
This leads to cURL call not being made.

Here is the `GetExpressCheckoutDetailsResponse` XML:
```
<?xml version="1.0" encoding="UTF-8"?>
<Response version="2">
  <PayPalTxn>
    <ack>Success</ack>
    <addressnormalizationstatus>None</addressnormalizationstatus>
    <amt>24.20</amt>
    <billing_agreement_accepted>0</billing_agreement_accepted>
    <build>38544305</build>
    <checkoutstatus>PaymentActionNotInitiated</checkoutstatus>
    <correlationid>a357f8827cc65</correlationid>
    <currencycode>EUR</currencycode>
    <custom>****</custom>
    <desc>*********</desc>
    <errorcode>0</errorcode>
    <handlingamt>0.00</handlingamt>
    <insuranceamt>0.00</insuranceamt>
    <insuranceoptionoffered>false</insuranceoptionoffered>
    <invnum>****</invnum>
    <shipdiscamt>0.00</shipdiscamt>
    <shippingamt>0.00</shippingamt>
    <taxamt>0.00</taxamt>
    <timestamp>2017-09-05T10:45:48Z</timestamp>
    <token>EC-92X45596YN1435128</token>
    <version>124.0</version>
  </PayPalTxn>
  <datacash_reference>3300900018396017</datacash_reference>
  <information>User has not yet accessed PayPal account. Please re-submit transaction.</information>
  <merchantreference>****_460_req</merchantreference>
  <mode>LIVE</mode>
  <reason>PayPal: Customer login pending</reason>
  <status>571</status>
  <time>1504608347</time>
</Response>
```

**Comment by Swedbank-SPP** [comment]
Hi, thanks. We've fixed this issue in v0.8.20 version.

**Comment by jsakars** [comment]
@Swedbank-SPP it does not seem to be fixed to me:

![image](https://user-images.githubusercontent.com/1607778/30201178-7213f89a-9482-11e7-9dce-a96fcfbf307b.png)

**Comment by Swedbank-SPP** [comment]
Hello, could you show a steps how do you initiate payment and what steps you do after success/return url, and exactly where this error occurs? During initPayment or somewhere else?

**Comment by jsakars** [comment]
Sure. These steps are the ones documented on SPP integration guide.

1. **Initialize the payment**
```
$response = $this->spp->getPayPalGateway()->initPayment(...);
return $response->getCustomerRedirectUrl($live);
```
User gets redirected to PayPal

2. Do nothing but **click `Cancel and return to ...` link**
![paypal](https://user-images.githubusercontent.com/1607778/30210924-560d17e2-94a7-11e7-9255-118a5a026c2d.png)

User gets redirected to `$cancelUrl`

3. **Call `handlePendingTransaction` method**
```
return $this->spp->getPayPalGateway()->handlePendingTransaction($merchantReference);
```
This is were Exception is thrown and therefore callback is never triggered.

![exception](https://user-images.githubusercontent.com/1607778/30211164-8b06add6-94a8-11e7-853d-e583df854387.png)

**Comment by Swedbank-SPP** [comment]
Thanks for detailed explanation. Please try now with updated v0.8.21 version.

**Comment by jsakars** [comment]
Seems we are going into the loop :)

![image](https://user-images.githubusercontent.com/1607778/30261156-e6d7ed46-96d3-11e7-990d-cad7767b30a3.png)

Callback is still not triggered but this time it comes from [try-catch](https://github.com/Swedbank-SPP/swedbank-payment-portal/blob/master/src/PayPal/PayPalService.php#L119-L136) which is ignoring the exception.

I wonder if You have actually tested this Yourself?

**Comment by Swedbank-SPP** [comment]
Please try v0.8.24 version.

**Comment by jsakars** [comment]
![image](https://user-images.githubusercontent.com/1607778/30265368-a6835c4c-96e4-11e7-9dc4-65dd1fd12471.png)

I don't want to be rude but this is getting frustrating - please test the code Yourself before releasing it!

**Comment by Swedbank-SPP** [comment]
Hi. Fixed this in v0.8.25

**Comment by jsakars** [comment]
Finally it works 😌
